pkg/oauth/apis/oauth/validation/validation.go

```diff
@@ -79,8 +89,24 @@ func ValidateAccessToken(accessToken *oauthapi.OAuthAccessToken) field.ErrorList
 
 func ValidateAccessTokenUpdate(newToken, oldToken *oauthapi.OAuthAccessToken) field.ErrorList {
 	allErrs := validation.ValidateObjectMetaUpdate(&newToken.ObjectMeta, &oldToken.ObjectMeta, field.NewPath("metadata"))
+	// since TimeoutsIn can be concurrently updated by multipe masters we do
+	// not allow it to decrease in value, as that would cause "updated" tokens
+	// to timeout earlier than they should. 0 is the exception, as this value
+	// indicates that the token does not expire, and it is an allowed new
+	// value.
+	if newToken.TimeoutsIn > 0 && newToken.TimeoutsIn < oldToken.TimeoutsIn {
```

Comment:
I had some related validation in `pkg/oauth/registry/oauthaccesstoken/strategy.go` that went with this as well.

Reply:
Yeah I saw that, but given we can update the field with any (valid) value, I do not see much value in adding this kind of validation just at token creation.

Reply:
OK I suppose that is fine.